src/bitcoin-cli.cpp

```diff
@@ -291,6 +293,184 @@ class GetinfoRequestHandler: public BaseRequestHandler
+        int ipv4_i{0}, ipv6_i{0}, onion_i{0}, block_relay_i{0}, total_i{0}; // inbound conn counters
+        int ipv4_o{0}, ipv6_o{0}, onion_o{0}, block_relay_o{0}, total_o{0}; // outbound conn counters
+        int max_peer_id_length{2}, max_version_length{1};
+        bool is_asmap_on{false};
+        std::vector<m_peer> peers;
+        const UniValue& getpeerinfo{batch[ID_PEERINFO]["result"]};
+        for (const UniValue& peer : getpeerinfo.getValues()) {
+            const std::string addr{peer["addr"].get_str()};
+            const std::string addr_local{peer["addrlocal"].isNull() ? "" : peer["addrlocal"].get_str()};
+            const int mapped_as{peer["mapped_as"].isNull() ? 0 : peer["mapped_as"].get_int()};
+            const bool is_block_relay{!peer["relaytxes"].get_bool()};
+            const bool is_inbound{peer["inbound"].get_bool()};
+            m_conn_type conn_type{m_conn_type::ipv4};
+            if (is_inbound) {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_i += 1;
+                } else if (IsInboundOnion(mapped_as, addr, addr_local)) {
+                    conn_type = m_conn_type::onion;
+                    onion_i += 1;
+                } else {
+                    ipv4_i += 1;
+                }
+                if (is_block_relay) block_relay_i += 1;
+            } else {
+                if (IsAddrIPv6(addr)) {
+                    conn_type = m_conn_type::ipv6;
+                    ipv6_o += 1;
+                } else if (IsOutboundOnion(addr)) {
+                    conn_type = m_conn_type::onion;
+                    onion_o += 1;
+                } else {
+                    ipv4_o += 1;
+                }
+                if (is_block_relay) block_relay_o += 1;
+            }
+            if (m_verbose) {
+                // Push data for this peer to the peers vector.
+                const int peer_id{peer["id"].get_int()};
+                const int version{peer["version"].get_int()};
+                const std::string sub_version{peer["subver"].get_str()};
+                const int64_t conn_time{peer["conntime"].get_int64()};
+                const int64_t last_recv{peer["lastrecv"].get_int64()};
+                const int64_t last_send{peer["lastsend"].get_int64()};
+                const double min_ping{peer["minping"].isNull() ? 0 : peer["minping"].get_real()};
+                const double ping{peer["pingtime"].isNull() ? 0 : peer["pingtime"].get_real()};
+                peers.push_back({peer_id, mapped_as, version, conn_time, last_recv, last_send, min_ping, ping, addr, sub_version, conn_type, is_block_relay, !is_inbound});
+
+                is_asmap_on |= (mapped_as != 0);
+                max_peer_id_length = std::max(int(ToString(peer_id).length()), max_peer_id_length);
+                max_version_length = std::max(int((ToString(version) + sub_version).length()), max_version_length);
```

Comment:
`std::string::length()` returns `size_t` and if our variables `max_peer_id_length` and `max_version_length` are defined as `size_t`, then there would not be a need for the typecast to `int`.

Reply:
Good idea